corehq/apps/hqwebapp/static/hqwebapp/js/mobile_experience_warning.js

```diff
@@ -1,40 +1,37 @@
 hqDefine('hqwebapp/js/mobile_experience_warning', function() {
-    if (window.innerWidth > 800 && window.innerHeight > 600) {
-        return;
-    }
+    var initialPageData = hqImport("hqwebapp/js/initial_page_data").get,
+        cookieName = "has-seen-mobile-experience-warning";
 
-    var cookieName = "has-seen-mobile-experience-warning";
-    if ($.cookie(cookieName)) {
-        return;
-    }
+    if (initialPageData("is_mobile_experience") && !$.cookie(cookieName)) {
```

Comment:
I'm not sure initial page data will work consistently outside of the document ready handler

Reply:
Also, as a nit: I usually try to avoid nesting too much, hence the organization above. That's definitely dealer's choice though.